expression/integration_test.go

```diff
@@ -4748,6 +4748,15 @@ func (s *testIntegrationSuite) TestFuncCaseWithLeftJoin(c *C) {
 	tk.MustQuery("select t1.id from kankan1 t1 left join kankan2 t2 on t1.id = t2.id where (case  when t1.name='b' then 'case2' when t1.name='a' then 'case1' else NULL end) = 'case1' order by t1.id").Check(testkit.Rows("1", "2"))
 }
 
+func (s *testIntegrationSuite) TestIssue11594(c *C) {
+	tk := testkit.NewTestKit(c, s.store)
+	tk.MustExec("use test")
+	tk.MustExec(`drop table if exists t1;`)
+	tk.MustExec("CREATE TABLE t1 (v bigint(20) UNSIGNED NOT NULL);")
+	tk.MustExec("INSERT INTO t1 VALUES (1), (2);")
+	tk.MustQuery("SELECT SUM(IF(v > 1, v, -v)) FROM t1;").Check(testkit.Rows("1"))
```

Comment:
Do we need to add some test cases for IFNULL, COALESCE?

Reply:
I find it's hard to construct cases for `IFNULL` and `COALESCE` 😥.
We need the first argument to be unsigned to let the result type of the multi-args function is unsigned, and trigger a converting from `unsigned int` to `decimal`.
Do you have any idea?

Reply:
SELECT sum(IFNULL(cast(null+rand() as unsigned), -v)) FROM t1;

Reply:
Nice.
A test for `IFNULL` is added.
And `COALESCE` doesn't have this problem since when we build a signature of `COALESCE`, we call `AggregateEvalType`, which takes the unsigned flag into account and amends the flag of the result type.

Reply:
We can add a test case to prevent unexpected changes by others.

Reply:
Okay, some tests for `COALESCE` have been added.